code/modules/mod/mod_control.dm

```diff
@@ -475,6 +477,23 @@
 	wearer.update_spacesuit_hud_icon("0")
 	wearer = null
 
+/obj/item/mod/control/proc/generate_suit_mask()
+	var/covered_slots = NONE
+	var/parts = get_parts(all = TRUE)
+	for(var/obj/item/part as anything in parts)
+		if(!get_part_datum(part).sealed)
+			parts -= part
+			continue
+		covered_slots |= part.slot_flags
+	if(suit_masks["[covered_slots]"])
+		return suit_masks["[covered_slots]"]
+	var/icon/slot_mask = icon('icons/blanks/32x32.dmi', "nothing")
```

Comment:
is 32x32 safe? will any of these ever be outside that bound?

Reply:
downstreams could fuck with it but nothing on tg should go outside that